walnuts_cpp/nuts.hpp

```diff
@@ -112,73 +112,13 @@ void leapfrog(const F& logp_grad_fun,
               Vec<S>& grad_next,
               S& logp_next) {
   S half_step = 0.5 * step;
-  rho_next = rho + half_step * grad;
-  theta_next = theta + step * (inv_mass.array() * rho_next.array()).matrix();
+  rho_next.noalias() = rho + half_step * grad;
+  theta_next.noalias() = theta + step * (inv_mass.array() * rho_next.array()).matrix();
```

Comment:
This is correct use of noalias. See the "Aliasing and matrix multiplication" section here 
https://eigen.tuxfamily.org/dox/group__TopicAliasing.html

Reply:
I think I was mistaken and it does do something here. I also think it’s safe to use even if it won’t lead to faster code

Reply:
I should get more fluent at building in godbolt.  Seriously, the config and dev ops is what's stopping me being an effective C++ programmer.  I find this stuff just totally overwhelming.